### ISSUE TITLE
Referals not working after recreating activity

### ISSUE THREAD

**Issue by Agraphie** [BranchMetrics/android-branch-deep-linking#495]
Hi there,

I'm using the 2.13.0 SDK and in general it works without any problems. However, if I recreate my activity branch referals are not working anymore until I kill the app. That means there are not params transmitted to the activity even though I clicked a branch link.

This is partly reproducible with the testbed App without changing anything except adding this in onStart() at the end

```Java
//Just some boolean to not recreate the activity all the time
if(!CustomBranchApp.recreated) {
    new Handler().postDelayed(new Runnable() {
        @Override
        public void run() {
            CustomBranchApp.recreated = true;
            recreate();
        }
    }, 5000);
}
````

In the testbed app the deep link activity won't start when you click the link. However, if you click it a second time then the activity will start again. This can also be achieved by simply launching the app, clicking a link and navigating back from the deep link activity. Now the link also has to be clicked twice in order to launch the deep link activity.

Do you have any ideas what might be happening? In general the deep linking works just not in that specific case. Any help is appreciated.

EDIT: With version 2.12.2 everything seems to work fine.

**Comment by sojanpr** [comment]
@Agraphie Glad to see that this is working fine with 2.12.2. We have added  a fix for some corner case intent handling when you launch the activity from history. Looks like that would fixed the issue

**Comment by sojanpr** [comment]
@Agraphie Please ignore the above comment since the issue was reported  on 2.13.0.
We are  working on this issue on 2.13.0 . Will update soon

**Comment by sojanpr** [comment]
@Agraphie  We have opened a PR https://github.com/BranchMetrics/android-branch-deep-linking/pull/497 for this issue. Will be releasing shortly after QA pass

**Comment by sojanpr** [comment]
@Agraphie  we have released v2.13.1 with the fix for this issue. Can you please try that.
